python/foglamp/services/core/scheduler/scheduler.py

```diff
@@ -1078,6 +1077,50 @@ def _schedule_row_to_schedule(cls,
             self._schedule_first_task(schedule_row, now)
             self._resume_check_schedules()
 
+    async def ping_service(self, service):
+        _MAX_ATTEMPTS = 5
+        """Number of max attempts for finding a heartbeat of service"""
+        attempt_count = 1
+        """Number of current attempt to ping url"""
+
+        url_ping = "{}://{}:{}/foglamp/service/ping".format(service._protocol,
+                                                            service._address,
+                                                            service._management_port)
+        async with aiohttp.ClientSession() as session:
+            while attempt_count < _MAX_ATTEMPTS + 1:
+                try:
+                    async with session.get(url_ping) as resp:
+                        res = await resp.json()
+                        if res["uptime"] is not None:
+                            break
+                except:
+                    attempt_count += 1
+                    time.sleep(1)
+            if attempt_count <= _MAX_ATTEMPTS:
+                self._logger.info('Ping received for Service %s id %s at url %s', service._name, service._id, url_ping)
+                return True
+        self._logger.error('Ping not received for Service %s id %s at url %s attempt_count %s', service._name, service._id,
+                      url_ping, attempt_count)
+        return False
+
+    async def shutdown_service(self, service):
+        _ping_timeout = 5  # type: int
+        """Timeout for a response from any given micro-service"""
+
+        try:
+            url_shutdown = "{}://{}:{}/foglamp/service/shutdown".format(service._protocol, service._address,
+                                                                        service._management_port)
+            async with aiohttp.ClientSession() as session:
+                async with session.post(url_shutdown) as resp:
+                    status_code = resp.status
+                    assert 200 == resp.status
```

Comment:
we should avoid assert practice in production code, you can keep else block with 200 check and return false and log error as in generic exception block

Reply:
Assertions are supposed to pass 100% of the time. An exception is only thrown when the condition unexpectedly fails. There's nothing wrong with running assertions in production code. They're not harmful, or bad design. They're good design, in fact. They confirm that the system is running correctly.